lib/utils.js

```diff
@@ -604,7 +604,10 @@ function decorateWithCollation(command, target, options) {
  * @param {object} command the command on which to apply the read concern
  * @param {Collection} coll the parent collection of the operation calling this method
  */
-function decorateWithReadConcern(command, coll) {
+function decorateWithReadConcern(command, coll, options) {
+  if (options && options.session && options.session.inTransaction()) {
```

Comment:
I think it's worth paying a lot of attention to this fix. It's not being done in the right place, but at least the application of a read concern has been isolated to a single place. I'd say just keep it in mind as we move towards the "command refactor"